exporter/elasticsearchexporter/internal/serializer/otelserializer/common.go

```diff
@@ -57,7 +57,18 @@ func writeAttributes(v *json.Visitor, attributes pcommon.Map, stringifyMapValues
 
 	_ = v.OnKey("attributes")
 	_ = v.OnObjectStart(-1, structform.AnyType)
+
+	// Deduplicate keys to workaround non-compliant SDKs.
+	// It is better to drop duplicate keys than to produce invalid JSON which may cause the entire document to be rejected.
+	seenKeys := make(map[string]struct{})
+
 	for k, val := range attributes.All() {
+		// Skip duplicate keys
+		if _, exists := seenKeys[k]; exists {
+			continue
```

Comment:
Maybe it would be great to have a debug log statement here for elements that are skipped. This could help cases where attributes have the same keys but different values - e.g. where attributes with the same keys are added by different components.

Reply:
> where attributes with the same keys are added by different components.

I see where you're coming from, but

- Unless the SDK is broken (like .NET https://github.com/open-telemetry/opentelemetry-dotnet/issues/4324), any keys added by different components should overwrite the key, instead of duplicating them.
- This is in alignment with `dedup` in ECS mapping mode, any duplicate keys in JSON (which in ECS's case is expected and not a bug), we do not have log statements in objmodel.go indicating any silent drops.
- If it was not a spec violation, I'd agree it'll be better to log it

Reply:
In a situation where a sidecar deployment or SDK sets a certain attribute and later it is also processed by a daemonset deployment, the same attribute key can be encountered with different values, as the implementations do have a different view on the system.